core/src/no/group15/playmagic/ecs/systems/MovementSystem.kt

```diff
@@ -49,6 +51,16 @@ class MovementSystem(
 				movement.moveRight -> transform.position = move(Input.Keys.RIGHT, transform.position)
 			}
 
+			if (gameMap.overlappingWithWall(entity)) {
```

Comment:
Shouldn't this be `if (!gameMap.overlappingWithWall(entity)) ...` so that you can only move when the `entity` is *not* overlapping with the wall?

Reply:
I move the player and then I check if it overlaps on a wall. Indeed the code inside the `if` is to revert the movement (probably the code is not clear right now, maybe we can implement it better with a `revert()` on the command pattern). You revert the movement only if the player went on a wall. If you do the check and then move the player, after the player go on a wall he can't move any more. Does this explanation clarify the logic? :slightly_smiling_face: